templates/decays.hpp.in

```diff
@@ -0,0 +1,276 @@
+// the generator is guessed from color representations of FieldIn, FieldOut1 and FieldOut2
+// This is not a bulletproof solution and might fail in general but is enough for
+// decays of color singlets
+template<typename FieldIn, typename FieldOut1, typename FieldOut2>
+double squared_color_generator() {
+
+   double color_factor;
+
+   if(@ModelName@_cxx_diagrams::fields::is_singlet<FieldIn>::value) {
+      // singlet to single singlet
+      if(@ModelName@_cxx_diagrams::fields::is_singlet<FieldOut1>::value && @ModelName@_cxx_diagrams::fields::is_singlet<FieldOut2>::value) {
+         color_factor = 1.;
+      }
+      // singlet to triplet and anti-triplet
+      else if((@ModelName@_cxx_diagrams::fields::is_triplet<FieldOut1>::value && @ModelName@_cxx_diagrams::fields::is_anti_triplet<FieldOut2>::value) ||
+            (@ModelName@_cxx_diagrams::fields::is_triplet<FieldOut2>::value && @ModelName@_cxx_diagrams::fields::is_anti_triplet<FieldOut1>::value)) {
+         color_factor = 3.;
+      }
+      // singlet to octet octet
+      else if(@ModelName@_cxx_diagrams::fields::is_octet<FieldOut1>::value && @ModelName@_cxx_diagrams::fields::is_octet<FieldOut2>::value) {
+         color_factor = 8.;
+      }
+      // singlet to unhandlet color combination
+      else {
+         throw std::runtime_error("Decay of color singlet to unhandled color combination");
+      }
+   }
+   else if(@ModelName@_cxx_diagrams::fields::is_octet<FieldIn>::value) {
+      if(@ModelName@_cxx_diagrams::fields::is_octet<FieldOut1>::value && @ModelName@_cxx_diagrams::fields::is_octet<FieldOut2>::value) {
+         color_factor = 40./3.;
+      }
+      // octet to triplet and anti-triplet
+      else if((@ModelName@_cxx_diagrams::fields::is_triplet<FieldOut1>::value && @ModelName@_cxx_diagrams::fields::is_anti_triplet<FieldOut2>::value) ||
+            (@ModelName@_cxx_diagrams::fields::is_triplet<FieldOut2>::value && @ModelName@_cxx_diagrams::fields::is_anti_triplet<FieldOut1>::value)) {
+         color_factor = 4.;
+      }
+      else if((@ModelName@_cxx_diagrams::fields::is_octet<FieldOut1>::value && @ModelName@_cxx_diagrams::fields::is_singlet<FieldOut2>::value) ||
+           (@ModelName@_cxx_diagrams::fields::is_singlet<FieldOut1>::value && @ModelName@_cxx_diagrams::fields::is_octet<FieldOut2>::value) ) {
+         color_factor = 8.;
+      }
+      // octet to unhandlet color combination
+      else {
+         throw std::runtime_error("Decay of color octet to unhandled color combination");
+      }
+   }
+   // decays of color charged particles are currently not supported
+   else {
+      throw std::runtime_error("Color factors for decays of color charged particles are not supported");
+   }
+
+   return color_factor;
+}
+
+template<typename FieldIn, typename FieldOut1, typename FieldOut2>
+std::string wrong_width_sign_msg(std::vector<int> in, std::vector<int> out1, std::vector<int> out2) {
```

Comment:
It seems to me that this function is model independent and may be better put into `src/`.

Reply:
Done in c95a39bd2d431d188b8347ce4342eded31d9ff2c.

Reply:
Thanks!